### PR TITLE
Dark mode: Color handling in the colors

### PR DESCRIPTION
**Pull request by louismaximepiton**
### Description

- A new `scss/_color-palette.scss` file has been created to define the color palette based on the one defined for Orange Design System: https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/blob/8ba87681fad206eb88cdab3c7d7b23cf9e5e70e5/scss/_color-palette.scss. This color palette adds various Sass variables named after the color names defined in the Orange Design System (e.g. `$ods-orange-200` or `$ods-forest-100`). Their values are set to hex values.
  - `_color-palette.scss` is imported by `_variables.scss` to avoid Sass compilation breaking changes.
  - It implies that `_variables.scss` and `_variables-dark.scss` now use these new variables instead of the hex values directly.
- The Bootstrap color palette for blues, indigos, purples, etc. has been mapped to the new Orange Design System color palette for the values from `${color}-100` to `${color}-600`. From `${color}-700` to `${color}-900`, `shade-color()` from `20%` to `60%` of the `${color}-600` value is used.

#### CSS Variables definition

- `--bs-placeholder-color` has been deleted and replaced by the direct use of `--bs-secondary-color`
- `--bs-disabled-color` has a new value: `--bs-tertiary-color`
- `--bs-active-bg` has been renamed `--bs-tertiary-active-bg`
- New `--bs-border-color-subtle` and `$border-color-subtle` to be used instead of `--bs-border-color-translucent` because our colors are never translucent. The translucent variable is kept for backward compatibility.
  - All the usage of `--bs-border-color-translucent` have been changed to `--bs-border-color-subtle` except when it wasn't used as a border color. In this case, it has been replaced with `--bs-tertiary-color`.
- `--bs-disabled-filter: #{$disabled-filter};` has been renamed to be more specific: `--bs-form-color-disabled-filter: #{$form-color-disabled-filter};`
- `--bs-hover-color` has been renamed to be more specific: `--bs-nav-link-active-color`
- `--bs-gray-tweak` has been renamed to be more specific: `--bs-form-check-input-disabled-color`
- `--bs-black-tweak` has been renamed to be more specific: `--bs-form-switch-square-bg`
- `--bs-input-filter` has been renamed to be more specific: `--bs-form-check-filter`
- `--bs-icon-filter` has been renamed to be more specific: `--breadcrumb-divider-filter`
- `--bs-select-disabled-indicator` and `--bs-select-indicator` has been renamed respectively to `--bs-form-select-disabled-indicator` and `--bs-form-select-indicator`

Note: here are saved values for `--bs-table-*-factor`:
```scss
--#{$prefix}table-striped-bg-factor: .135; // .2 if solid bg
--#{$prefix}table-striped-hover-bg-factor: .855; // .865 if solid bg
--#{$prefix}table-hover-bg-factor: .135; // .2 if solid bg
--#{$prefix}table-active-bg-factor: .565; // .6 if solid bg
```

#### Variables usage

- `$brand-orange` is replaced with `$supporting-orange`
- `$accessible-orange` is replaced with `$primary`
- `--bs-highlight-bg` is replaced with `--secondary`, `--bs-border-color` or `--bs-body-color` given the context
- `--bs-link-hover-color` is replaced with `--bs-primary` for the cases where it's not used as a link

#### Documentation

- Color modes page has been updated to:
  - Drop the design callout
  - Adapt the dropdown example description
  - Add a warning callout about the impact of deactivating entirely the dark mode
- New color palette page
- New color theme page
- Deletion of color page

#### Examples

- Download page has been updated to use the new CSS variables. Impact for users in the main branch?

### Previews

- https://deploy-preview-2382--boosted.netlify.app/docs/5.3/customize/color/
- https://deploy-preview-2382--boosted.netlify.app/docs/5.3/customize/color-theme/
- https://deploy-preview-2382--boosted.netlify.app/docs/5.3/customize/color-palette/
- https://deploy-preview-2382--boosted.netlify.app/docs/5.3/customize/color-modes/

### CODE REVIEW

**Review by louismaximepiton — COMMENTED** [review]
- `$brand-orange` -> `$supporting-orange`
- New Boosted palette Sass variables
- Some CSS var replacement/renamed/new:
  - `--bs-active-bg` -> `--bs-tertiary-active-bg`
  - `--bs-border-color-translucent` -> `--bs-border-color-subtle`
  - `--bs-input-filter` -> `--bs-form-check-filter`
  - `--bs-gray-tweak` -> `--bs-form-check-input-disabled-color`
  - `--bs-disabled-filter` -> `--bs-form-color-disabled-filter` + removal of one Sass variable
  - `--bs-select-indicator` -> `--bs-form-select-indicator`
  - `--bs-select-disabled-indicator` -> `--bs-form-select-disabled-indicator`
  - `--bs-black-tweak` -> `--bs-form-switch-square-bg`
  - `--bs-icon-filter` -> `--bs-breadcrumb-divider-filter`
  - `--bs-hover-color` -> `--bs-nav-link-active-color`
  - `--bs-placeholder-color` -> `--bs-secondary-color`
  - `--bs-body-color-subtle` -> `-`
- `--bs-code-color` change its color on dark mode
- `--bs-secondary-color` change its color on dark mode
- `--bs-tertiary-color` change its color on light/dark mode
- All the `$color-*` changed

I still have some files to review but here is a first part

File: site/content/docs/5.3/customize/color.md
Comment (julien-deramond) [inline site/content/docs/5.3/customize/color.md:1]: As a reminder, double-check that a redirection exists for this page

File: site/data/palette.yml
Comment (julien-deramond) [inline site/data/palette.yml:1]: As a reminder, double-check that everything in this file is actually used for the documentation

File: site/assets/js/color.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+// NOTICE!! DO NOT USE ANY OF THIS JAVASCRIPT
+// IT'S ALL JUST JUNK FOR OUR DOCS!
+// ++++++++++++++++++++++++++++++++++++++++++
+
+/*!
+ * JavaScript for Bootstrap's docs (https://getbootstrap.com/)
+ * Copyright 2011-2023 The Bootstrap Authors
```
[the remark this answers is not in the document]
Comment (julien-deramond) [reply site/assets/js/color.js:7]: I dropped the header

File: site/data/palette.yml [hunk 2]
```diff
@@ -1,22 +1,21 @@
 - category: Core colors
   name: "Core colors"
   colors:
-    - name: White
-      class: white
-      hex: "#fff"
-      variable: $white
+    - name: Orange
+      class: primary
```
Comment (julien-deramond) [inline site/data/palette.yml:5]: Not entirely sure, but `class` isn't used anymore in this file, isn't it?

File: site/data/palette.yml [hunk 3]
```diff
@@ -40,64 +39,249 @@
       class: pink
       hex: "#ffb4e6"
       variable: $supporting-pink
```
Comment (julien-deramond) [inline site/data/palette.yml:41]: The `$supporting-*` variables aren't used anymore isn't it?

File: site/data/palette.yml
Comment (julien-deramond) [inline site/data/palette.yml:1]: There are things to fine-tune in this file. Some elements seem not to be used anymore:
- "Core colors" -> `class`
- "Supporting colors" -> `variable`
- etc.

**Pull request merged**